## Hypothesis Testing

In [27]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv('C:/Users/divyanshu/OneDrive/Desktop/Projects/ImpactSense/impactsenseAI-Infosys-Intern-project/data/earthquakes_data.csv')

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [28]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0734562488664734
p-value: 0.28557415327561997


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [29]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.51475865746147
p-value: 8.93302171443802e-112


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [30]:
# Bin magnitudes into categories
df['time'] = pd.to_datetime(df['time'], errors='coerce')

# Extract Month name or number
df['Month'] = df['time'].dt.month_name() 

df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 52.60608192508434
Degrees of freedom: 22
p-value: 0.00025767587156685375


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [31]:

df['DayOfWeek'] = df['time'].dt.day_name()
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.16105584276141
Degrees of freedom: 36
p-value: 8.095901679664044e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [32]:
df['time'] = pd.to_datetime(df['time'], errors='coerce')

df['Year'] = df['time'].dt.year
df['Month'] = df['time'].dt.month
df['DayOfWeek'] = df['time'].dt.day_name()


df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416162
p-value: 2.039284602025331e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

#### 1. Earthquake Magnitude and Geographic Distribution
Hypothesis: There is a significant difference in earthquake magnitudes between events that occurred near the equator (low latitudes) and those farther from it (high latitudes).

In [33]:
near_eq = df[df['latitude'].abs() <= 10]['mag'].dropna()
far_eq = df[df['latitude'].abs() > 10]['mag'].dropna()

ttest_result2 = ttest_ind(near_eq, far_eq, equal_var=False)
print('t-test statistic:', ttest_result2.statistic)
print('p-value:', ttest_result2.pvalue)

t-test statistic: -8.487840895465084
p-value: 2.15835652383957e-17


**Statement**: If the t-test yields a p-value > 0.05, we fail to reject the null hypothesis, suggesting that earthquake magnitudes are not significantly different between near-equatorial and higher-latitude regions.
If the p-value < 0.05, we reject the null hypothesis, indicating that earthquakes near the equator tend to have significantly different magnitudes compared to those farther away.

#### 2. Relationship Between Earthquake Depth and Magnitude Category
Hypothesis: There is an association between the depth of an earthquake and its magnitude category (Low, Medium, High, Very High).

In [34]:
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])
df['depth_bin'] = pd.cut(df['depth'], bins=[0, 70, 300, 700], labels=['Shallow', 'Intermediate', 'Deep'])

# Create contingency table
contingency = pd.crosstab(df['mag_bin'], df['depth_bin'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 222.72298569413664
Degrees of freedom: 4
p-value: 4.863351513571374e-47


**Statement** : If the p-value < 0.05, there is a statistically significant relationship between earthquake depth and magnitude — suggesting that deeper earthquakes may differ in intensity compared to shallow ones.If p-value > 0.05, we fail to reject the null hypothesis, indicating no strong association between an earthquake’s depth and its magnitude category.

#### 3. Relationship Between Location Longitude and Earthquake Magnitude
Hypothesis: Earthquakes occurring in eastern longitudes (positive longitude) differ significantly in magnitude from those in western longitudes (negative longitude).

In [35]:
east = df[df['longitude'] > 0]['mag'].dropna()
west = df[df['longitude'] < 0]['mag'].dropna()

ttest_result5 = ttest_ind(east, west, equal_var=False)
print('t-test statistic:', ttest_result5.statistic)
print('p-value:', ttest_result5.pvalue)


t-test statistic: -2.5101239140771905
p-value: 0.012070786094274422


**Statement**: If the p-value < 0.05, we can say there is a statistically significant difference in the mean earthquake magnitude between the eastern and western hemispheres.
If p-value > 0.05, we conclude that earthquake magnitudes are not significantly different across hemispheres, implying global uniformity in seismic intensity.

#### 4. Association Between Event Type and Review Status
Hypothesis: There is a significant association between the earthquake type (e.g., earthquake, quarry blast, explosion) and whether the event was reviewed or not.

In [36]:
contingency = pd.crosstab(df['type'], df['status'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)


Chi-square statistic: 0.467127846951434
Degrees of freedom: 6
p-value: 0.9982158170618664


**Statement**: If the p-value < 0.05, it suggests that the likelihood of an event being reviewed depends on its type — for example, true earthquakes might be reviewed more often than artificial explosions.If p-value > 0.05, we infer that event type and review status are independent, indicating no bias or dependency in the review process.

#### 5. Magnitude Difference Between Daytime and Nighttime Earthquakes
Hypothesis: The mean magnitude of earthquakes occurring during the day differs significantly from those occurring at night.

In [37]:
# Convert time column to datetime and extract hour
df['time'] = pd.to_datetime(df['time'], errors='coerce')
df['hour'] = df['time'].dt.hour

# Define daytime (6 AM–6 PM) and nighttime (6 PM–6 AM)
daytime = df[(df['hour'] >= 6) & (df['hour'] < 18)]['mag'].dropna()
nighttime = df[(df['hour'] < 6) | (df['hour'] >= 18)]['mag'].dropna()

ttest_result6 = ttest_ind(daytime, nighttime, equal_var=False)
print('t-test statistic:', ttest_result6.statistic)
print('p-value:', ttest_result6.pvalue)


t-test statistic: 0.6811841471655634
p-value: 0.4957564100106169


**Statement**: If the p-value < 0.05, we reject the null hypothesis — indicating that daytime and nighttime earthquakes differ in average magnitude.
If p-value > 0.05, we fail to reject the null hypothesis — meaning earthquake magnitudes do not vary significantly with time of day.